asdf/tags/transform/compound.py

```diff
@@ -87,7 +87,7 @@ def _to_tree_from_model_tree(cls, tree, ctx):
         except KeyError:
             raise ValueError("Unknown operator '{0}'".format(tree.value))
 
-        node = tagged.tag_object(cls.make_yaml_tag(tag_name), node)
+        node = tagged.tag_object(cls.make_yaml_tag(tag_name), node, ctx=ctx)
```

Comment:
Hm. The `CompoundType` represents a compound model of type `astropy.modeling.core.CompoundModel`. In my experience it never fails serialization. Why does it need to be wrapped? I may be wrong but my understanding is that the intention was to tag only python primitive types.
It would really help if there's a failing example which illustrates the problem.